lib/active_model/serializer.rb

```diff
@@ -105,7 +105,7 @@ def self.root_name
       name.demodulize.underscore.sub(/_serializer$/, '') if name
     end
 
-    attr_accessor :object, :root, :meta, :meta_key, :scope
+    attr_accessor :object, :root, :meta, :meta_key, :links, :scope
```

Comment:
@bf4 thanks for clarification :+1: 

The way I did was to conform with `meta` already present on the code. But as you said, and I agree, it should be  moved to the `adapter`. I'll update the code as you suggested.

Another issue is that both PR (mine and #1041) are defining the attribute `links` of the `@hash` 

See [here](https://github.com/rails-api/active_model_serializers/pull/1018/files#diff-ae1e38e96a8bf28ae37918a65c448c73R34) and [here](https://github.com/rails-api/active_model_serializers/pull/1041/files#diff-ae1e38e96a8bf28ae37918a65c448c73R167)

What's the best way to avoid conflict ? Just check if `links` is already present and append to it ? 

```ruby
render json: @posts, pagination: true
```

```json
{
  "data": [
    { }
  ],
  "links": {
    "first": "http://example.com/articles?page[number]=1&page[size]=1",
    "prev": "http://example.com/articles?page[number]=2&page[size]=1",
    "next": "http://example.com/articles?page[number]=4&page[size]=1",
    "last": "http://example.com/articles?page[number]=13&page[size]=1"
  }
}
```

```ruby
render json: @posts,
           pagination: true,
           links: { 
               "self": "http://example.com/articles",
               "last": "http://example.com/articles/last"
           }
```

```json
{
  "data": [
    { }
  ],
  "links": {
    "self": "http://example.com/articles",
    "first": "http://example.com/articles?page[number]=1&page[size]=1",
    "prev": "http://example.com/articles?page[number]=2&page[size]=1",
    "next": "http://example.com/articles?page[number]=4&page[size]=1",
    "last": "http://example.com/articles/last"
  }
}
```

See that I replaced `last` from `links`. Is that the expected output, or should I conserve attributes already defined ?